cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/services/http/handlers/AppLifecycleHttpHandlerTest.java

```diff
@@ -141,16 +140,16 @@ public void testOwnerUsingArtifact() throws Exception {
     addAppArtifact(artifactId.toId(), WordCountApp.class);
     ApplicationId applicationId = new ApplicationId(NamespaceId.DEFAULT.getNamespace(), "WordCountApp");
     // deploy an app with a owner
-    KerberosPrincipalId ownerPrincipal = new KerberosPrincipalId("alice/somehost.net@somekdc.net");
+    String ownerPrincipal = "alice/somehost.net@somekdc.net";
     AppRequest<ConfigTestApp.ConfigClass> appRequest = new AppRequest<>(
       new ArtifactSummary(artifactId.getArtifact(), artifactId.getVersion()), null, ownerPrincipal);
     Assert.assertEquals(HttpResponseCodes.SC_OK, deploy(applicationId, appRequest).getStatusLine().getStatusCode());
 
     // should be able to retrieve the owner information of the app
     JsonObject appDetails = getAppDetails(NamespaceId.DEFAULT.getNamespace(), applicationId.getApplication());
-    Assert.assertEquals(ownerPrincipal.getPrincipal(),
-                        appDetails.get(Constants.Security.OWNER_PRINCIPAL).
-                          getAsJsonObject().get("principal").getAsString());
+    Assert.assertEquals(ownerPrincipal,
```

Comment:
so the app detail still has the principal as a struct instead of a string? 
you changed that in other places, why not here?